feat: add support for Elasticache global replication group

apavamontri opened the pull request:
<!--
Please read through https://git.io/fj2m9 if this is your first time opening a
pull request to this repo. Find us in https://crossplane.slack.com
if you need any help contributing.
-->

### Description of your changes

<!--
Briefly describe what this pull request does. Be sure to direct your
reviewers' attention to anything that needs special consideration.

We love pull requests that resolve an open issue. If yours does, you
can uncomment the below line to indicate which issue your PR fixes, for example
"Fixes #500":
-->

Fixes #1333. Implementing `aws_elasticache_global_replication_group`

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change.
-->

Running Uptest locally.

```text
--- PASS: kuttl (1826.27s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/case (1825.48s)
PASS
15:24:48 [ OK ] running automated tests
```

Also testing the following cluster setup:
* `auth_token`: reading from Kubernetes secret
* multi-az, failover, number of clusters > 1
* in-transit encryption

Timeout need to increase to 1 hour due the at least two clusters creation using `uptest.upbound.io/timeout: "3600"` annotation.

[contribution process]: https://git.io/fj2m9

turkenf commented:
/test-examples="examples/elasticache/v1beta1/globalreplicationgroup.yaml"

turkenf commented on config/externalname.go:
```diff
@@ -1141,12 +1141,13 @@ var TerraformPluginSDKExternalNameConfigs = map[string]config.ExternalName{
-	"aws_elasticache_user":              config.ParameterAsIdentifier("user_id"),
-	"aws_elasticache_user_group":        config.ParameterAsIdentifier("user_group_id"),
+	"aws_elasticache_cluster":                  config.ParameterAsIdentifier("cluster_id"),
+	"aws_elasticache_global_replication_group": config.IdentifierFromProvider,
```
nit: it would be great to add a comment line about import

```suggestion
        // ElastiCache Global Replication Groups can be imported using the global_replication_group_id
	"aws_elasticache_global_replication_group": config.IdentifierFromProvider,
```

turkenf commented on config/externalnamenottested.go:
```diff
@@ -219,7 +219,7 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
 	// which is an attribute reported in the state.
 	// TODO: we need to check the value of a global_replication_group_id to
 	// see if further normalization is possible
-	"aws_elasticache_global_replication_group": config.IdentifierFromProvider,
+	// "aws_elasticache_global_replication_group": config.IdentifierFromProvider,
```
As I can see in the uptest run, the `global_replication_group_id` is the suffix we provided `example` and contains the `virxk` value generated by the cloud provider: `virxk-example`. In this case, normalization is not possible, please remove all lines related to the GlobalReplicationGroup resource (including line 218) .

apavamontri replied:
Remove commented code

turkenf reviewed (commented):
Thank you for this dedicated PR @apavamontri. I left two small comments. 

Could you please squash your commits to make the commit history cleaner?

turkenf commented:
And please sign your commits: https://github.com/crossplane-contrib/provider-upjet-aws/pull/1485/checks?check_run_id=29807125527

apavamontri commented:
> Thank you for this dedicated PR @apavamontri. I left two small comments.
> 
> Could you please squash your commits to make the commit history cleaner?

Rebased and squashed.  Thank you for your patient.

apavamontri commented:
> And please sign your commits: https://github.com/crossplane-contrib/provider-upjet-aws/pull/1485/checks?check_run_id=29807125527

@turkenf, It was my mistake. Add `--signoff` during rebased and squashed. Thank you for your patience and for helping out during this process

turkenf commented:
/test-examples="examples/elasticache/v1beta1/globalreplicationgroup.yaml"

turkenf reviewed (approved):
Thanks a lot @apavamontri, LGTM 🙌
